OptimoveSDK/Tests/Sources/Optimobile/AnalyticsHelperTests.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  Test.swift
+//  Optimove
+//
+//  Created by Kostya Antipochkin on 2024-11-22.
+//
+
+import XCTest
+@testable import OptimoveSDK
+import OptimoveTest
+
+class AnalyticsHelperTests: XCTestCase {
+    
+    var mockHttpClient: MockKSHttpClient!
+    var analyticsHelper: AnalyticsHelper!
+    
+    override func setUp() {
+        super.setUp()
+        mockHttpClient = MockKSHttpClient()
+        analyticsHelper = AnalyticsHelper(httpClient: mockHttpClient)
+    }
+    
+    override func tearDown()
+    {
+        analyticsHelper = nil
+        mockHttpClient = nil
+        super.tearDown()
+    }
+    
+    func test_number_of_sent_events_same_as_tracked() {
+        let numberOfEvents = 8;
+        let numberOfEventsExpectation = expectation(description: "Number of events wasnt \(numberOfEvents)")
+        
+        mockHttpClient.forward = { method, data in
+            if let data = data as? [[String: Any?]], data.count == numberOfEvents {
+                numberOfEventsExpectation.fulfill()
+            }
+        }
+        
+        for i in 1...numberOfEvents {
+            self.analyticsHelper.trackEvent(eventType: "immediate_event\(i)", properties: nil, immediateFlush: true)
```

Comment:
do we need `self` in `analyticsHelper.trackEvent`? I know it's usually optional, but wondering if there is a reason here

Reply:
Removed